lib/kube/proxy/cluster_details.go

```diff
@@ -328,51 +340,79 @@ func getAWSResourceMatcherToCluster(kubeCluster types.KubeCluster, resourceMatch
 		if match, _, _ := services.MatchLabels(matcher.Labels, kubeCluster.GetAllLabels()); !match {
 			continue
 		}
-
-		return &(matcher.AWS)
+		return &matcher.AWS
 	}
 	return nil
 }
 
+// STSPresignClient is the subset of the STS presign interface we use in fetchers.
+type STSPresignClient = kubeutils.STSPresignClient
+
+// EKSClient is the subset of the EKS Client interface we use.
+type EKSClient interface {
+	eks.DescribeClusterAPIClient
+}
+
+// STSClient is the subset of the STS Client interface we use.
+type STSClient interface {
+	stscreds.AssumeRoleAPIClient
+}
+
+// ClientGetter is an interface for getting an EKS client and an STS client.
+type ClientGetter interface {
+	// GetAWSEKSClient returns AWS EKS client for the specified region.
+	GetAWSEKSClient(ctx context.Context, region string, opts ...awsconfig.OptionsFn) (EKSClient, error)
+	// GetAWSSTSClient returns AWS STS client for the specified region.
+	GetAWSSTSClient(ctx context.Context, region string, opts ...awsconfig.OptionsFn) (STSClient, error)
+	// GetAWSSTSPresignClient returns AWS STS presign client for the specified region.
+	GetAWSSTSPresignClient(ctx context.Context, region string, opts ...awsconfig.OptionsFn) (STSPresignClient, error)
+}
+
 // getAWSClientRestConfig creates a dynamicCredsClient that generates returns credentials to EKS clusters.
-func getAWSClientRestConfig(cloudClients cloud.Clients, clock clockwork.Clock, resourceMatchers []services.ResourceMatcher) dynamicCredsClient {
+func getAWSClientRestConfig(cloudClients ClientGetter, clock clockwork.Clock, resourceMatchers []services.ResourceMatcher) dynamicCredsClient {
 	return func(ctx context.Context, cluster types.KubeCluster) (*rest.Config, time.Time, error) {
 		region := cluster.GetAWSConfig().Region
-		opts := []cloud.AWSOptionsFn{
-			cloud.WithAmbientCredentials(),
-			cloud.WithoutSessionCache(),
+		opts := []awsconfig.OptionsFn{
+			awsconfig.WithAmbientCredentials(),
+			// TODO(@GavinFrazar): Re-enable this when session cache v2 gets merged (#50561).
+			// awsconfig.WithoutSessionCache(),
```

Comment:
we need this merged otherwise we will introduce the same bug https://github.com/gravitational/teleport/issues/50074 again

Reply:
Done.